src/QA/QA.Web/Controllers/WithdrawVacancyController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.ComponentModel;
+using System.Threading.Tasks;
+using Esfa.Recruit.Qa.Web.Configuration.Routing;
+using Esfa.Recruit.Qa.Web.Extensions;
+using Esfa.Recruit.Qa.Web.Models.WithdrawVacancy;
+using Esfa.Recruit.Qa.Web.Orchestrators;
+using Esfa.Recruit.Qa.Web.Security;
+using Esfa.Recruit.Qa.Web.ViewModels.WithdrawVacancy;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Esfa.Recruit.Qa.Web.Controllers
+{
+    [Authorize(Policy = AuthorizationPolicyNames.TeamLeadUserPolicyName)]
+    [Route(RoutePaths.WithdrawVacancyPath)]
+    public class WithdrawVacancyController : Controller
+    {
+        private readonly WithdrawVacancyOrchestrator _orchestrator;
+
+        public WithdrawVacancyController(WithdrawVacancyOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+        }
+
+        [HttpGet(Name = RouteNames.WithdrawVacancy_FindVacancy_Get)]
+        public IActionResult FindVacancy()
+        {
+            var vm = _orchestrator.GetFindVacancyViewModel();
+
+            return View(vm);
+        }
+
+        [HttpPost(Name = RouteNames.WithdrawVacancy_FindVacancy_Post)]
+        public async Task<IActionResult> FindVacancy(FindVacancyEditModel m)
+        {
+            if (ModelState.IsValid)
+            {
+                var result = await _orchestrator.PostFindVacancyEditModelAsync(m);
+
+                switch (result.ResultType)
+                {
+                    case PostFindVacancyEditModelResultType.AlreadyClosed:
+                        return RedirectToRoute(RouteNames.WithdrawVacancy_AlreadyClosed_Get, new {vacancyReference = result.VacancyReference});
+                    case PostFindVacancyEditModelResultType.CanClose:
+                        return RedirectToRoute(RouteNames.WithdrawVacancy_Confirm_Get, new {vacancyReference = result.VacancyReference});
+                    case PostFindVacancyEditModelResultType.NotFound:
+                        ModelState.AddModelError(nameof(FindVacancyViewModel.VacancyReference), $"Cannot find a live vacancy with vacancy reference '{m.VacancyReference}'");
+                        break;
+                    default:
+                        throw new InvalidEnumArgumentException($"{result.ResultType.ToString()} is not handled");
```

Comment:
InvalidEnumArgumentException should be thrown when passing an invalid enum value to a method or a property setter.
Your code is encountering a valid enum that it doesn't implement, so throw a NotImplementedException.

Reply:
👍